cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -2077,30 +2077,62 @@ pInvalidBefore = fmap SlotNo $ asum
     ]
   ]
 
-pInvalidHereafter :: Parser SlotNo
-pInvalidHereafter =
+pLegacyInvalidHereafter :: Parser SlotNo
```

Comment:
But we also have legacy commands that don't understand eras the same way.  In the legacy era, options are always there regardless of era and these need validation.  This is why these parse plain types, not era sensitive types.

This means we may receive values from the CLI parser that are invalid for our era and hence we need validation.